pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -16,6 +16,22 @@
     timedelta_range,
     Timedelta, Timestamp, NaT, Series, TimedeltaIndex, DatetimeIndex,
     DataFrame)
+from pandas.core.arrays import (
+    TimedeltaArrayMixin as TimedeltaArray,
+    DatetimeArrayMixin as DatetimeArray)
+
+
+def get_upcast_box(box, vector):
```

Comment:
Eventually we'll want something like this, probably put it next to `tm.box_expected`.  After this I'll end up rebasing #23734 and see if this can be re-used in the other files.  There are enough special cases that it isn't obvious off the top of my head.

Reply:
did you rebase off of #23734? isn't this basically box_expected with another arg? there are a LOT of helpers out there. too many.....

Reply:
Yes this has been rebased.  This is about determining what box to use when we are parametrizing over two box types.  I agree it may be nice to cut down on the helpers; in fact I think we may be able to get rid of `tm.box_expected` if we're lucky (fixing the transpose thing...)